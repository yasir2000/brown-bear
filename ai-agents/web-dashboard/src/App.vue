<template>
  <div class="min-h-screen bg-gray-50">
    <!-- Navigation -->
    <nav class="bg-white shadow-sm">
      <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
        <div class="flex justify-between h-16">
          <div class="flex">
            <div class="flex-shrink-0 flex items-center">
              <h1 class="text-2xl font-bold text-indigo-600">ðŸ¤– Brown Bear AI</h1>
            </div>
            <div class="hidden sm:ml-6 sm:flex sm:space-x-8">
              <router-link to="/" class="nav-link">Dashboard</router-link>
              <router-link to="/agents" class="nav-link">Agents</router-link>
              <router-link to="/tasks" class="nav-link">Tasks</router-link>
              <router-link to="/analytics" class="nav-link">Analytics</router-link>
            </div>
          </div>
          <div class="flex items-center">
            <span class="px-3 py-1 rounded-full text-sm font-medium bg-green-100 text-green-800">
              {{ connectionStatus }}
            </span>
          </div>
        </div>
      </div>
    </nav>

    <!-- Main Content -->
    <main class="max-w-7xl mx-auto py-6 sm:px-6 lg:px-8">
      <router-view />
    </main>
  </div>
</template>

<script setup lang="ts">
import { ref, onMounted } from 'vue';
import { useAgentStore } from './stores/agents';

const agentStore = useAgentStore();
const connectionStatus = ref('Connected');

onMounted(async () => {
  await agentStore.fetchAgents();
});
</script>

<style scoped>
.nav-link {
  @apply inline-flex items-center px-1 pt-1 border-b-2 text-sm font-medium transition-colors;
  @apply border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300;
}

.nav-link.router-link-active {
  @apply border-indigo-500 text-gray-900;
}
</style>
