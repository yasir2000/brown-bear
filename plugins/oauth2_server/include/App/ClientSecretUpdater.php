<?php
/**
 * Copyright (c) Enalean, 2020-Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

declare(strict_types=1);

namespace Tuleap\OAuth2Server\App;

use Tuleap\Authentication\SplitToken\SplitTokenVerificationString;
use Tuleap\Authentication\SplitToken\SplitTokenVerificationStringHasher;
use Tuleap\OAuth2ServerCore\App\AppDao;

class ClientSecretUpdater
{
    /**
     * @var SplitTokenVerificationStringHasher
     */
    private $hasher;
    /**
     * @var AppDao
     */
    private $app_dao;
    /**
     * @var LastGeneratedClientSecretStore
     */
    private $client_secret_store;

    public function __construct(
        SplitTokenVerificationStringHasher $hasher,
        AppDao $app_dao,
        LastGeneratedClientSecretStore $client_secret_store,
    ) {
        $this->hasher              = $hasher;
        $this->app_dao             = $app_dao;
        $this->client_secret_store = $client_secret_store;
    }

    public function updateClientSecret(int $app_id): void
    {
        $secret        = SplitTokenVerificationString::generateNewSplitTokenVerificationString();
        $hashed_secret = $this->hasher->computeHash($secret);
        $this->app_dao->updateSecret($app_id, $hashed_secret);
        $this->client_secret_store->storeLastGeneratedClientSecret($app_id, $secret);
    }
}
