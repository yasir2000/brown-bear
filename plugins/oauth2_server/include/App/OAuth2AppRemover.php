<?php
/**
 * Copyright (c) Enalean, 2020-Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

declare(strict_types=1);

namespace Tuleap\OAuth2Server\App;

use Tuleap\DB\DBTransactionExecutor;
use Tuleap\OAuth2ServerCore\Grant\AuthorizationCode\OAuth2AuthorizationCodeDAO;
use Tuleap\OAuth2Server\User\AuthorizationDao;
use Tuleap\OAuth2ServerCore\App\AppDao;

class OAuth2AppRemover
{
    /**
     * @var AppDao
     */
    private $app_dao;
    /**
     * @var OAuth2AuthorizationCodeDAO
     */
    private $authorization_code_dao;
    /**
     * @var AuthorizationDao
     */
    private $authorization_dao;
    /**
     * @var DBTransactionExecutor
     */
    private $transaction_executor;

    public function __construct(
        AppDao $app_dao,
        OAuth2AuthorizationCodeDAO $authorization_code_dao,
        AuthorizationDao $authorization_dao,
        DBTransactionExecutor $transaction_executor,
    ) {
        $this->app_dao                = $app_dao;
        $this->authorization_code_dao = $authorization_code_dao;
        $this->authorization_dao      = $authorization_dao;
        $this->transaction_executor   = $transaction_executor;
    }

    public function deleteAppByID(int $app_id): void
    {
        $this->transaction_executor->execute(
            function () use ($app_id): void {
                $this->app_dao->delete($app_id);
                $this->authorization_code_dao->deleteAuthorizationCodeByAppID($app_id);
                $this->authorization_dao->deleteAuthorizationByAppID($app_id);
            }
        );
    }
}
