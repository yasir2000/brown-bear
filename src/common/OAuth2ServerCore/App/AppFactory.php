<?php
/**
 * Copyright (c) Enalean, 2022-Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

declare(strict_types=1);

namespace Tuleap\OAuth2ServerCore\App;

use ProjectManager;

class AppFactory
{
    public function __construct(private RetrieveAppMatchingClientID $app_retriever, private ProjectManager $project_manager)
    {
    }

    /**
     * @throws OAuth2AppNotFoundException
     */
    public function getAppMatchingClientId(ClientIdentifier $client_identifier): OAuth2App
    {
        $row = $this->app_retriever->searchByClientId($client_identifier);
        if (! $row) {
            throw new OAuth2AppNotFoundException($client_identifier);
        }
        $project = null;
        if ($row['project_id'] !== null) {
            try {
                $project = $this->project_manager->getValidProject($row['project_id']);
            } catch (\Project_NotFoundException $e) {
                throw new OAuth2AppNotFoundException($client_identifier);
            }
        }
        return new OAuth2App($row['id'], $row['name'], $row['redirect_endpoint'], (bool) $row['use_pkce'], $project);
    }
}
