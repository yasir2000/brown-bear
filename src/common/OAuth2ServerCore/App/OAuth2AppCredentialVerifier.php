<?php
/**
 * Copyright (c) Enalean, 2020-Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

declare(strict_types=1);

namespace Tuleap\OAuth2ServerCore\App;

use Tuleap\Authentication\SplitToken\SplitToken;
use Tuleap\Authentication\SplitToken\SplitTokenVerificationStringHasher;

class OAuth2AppCredentialVerifier
{
    /**
     * @var AppFactory
     */
    private $app_factory;
    /**
     * @var AppDao
     */
    private $app_dao;
    /**
     * @var SplitTokenVerificationStringHasher
     */
    private $hasher;

    public function __construct(AppFactory $app_factory, AppDao $app_dao, SplitTokenVerificationStringHasher $hasher)
    {
        $this->app_factory = $app_factory;
        $this->app_dao     = $app_dao;
        $this->hasher      = $hasher;
    }

    /**
     * @throws OAuth2AppNotFoundException
     * @throws OAuth2ClientIdentifierAndSecretMismatchException
     * @throws InvalidOAuth2AppSecretException
     * @throws OAuth2MissingVerifierStringException
     */
    public function getApp(ClientIdentifier $client_identifier, SplitToken $client_secret): OAuth2App
    {
        $app = $this->app_factory->getAppMatchingClientId($client_identifier);

        if ($app->getId() !== $client_secret->getID()) {
            throw new OAuth2ClientIdentifierAndSecretMismatchException($client_identifier);
        }

        $client_verification_expected_string = $this->app_dao->searchClientSecretByClientID($app->getId());
        if ($client_verification_expected_string === null) {
            throw new OAuth2MissingVerifierStringException($app);
        }

        $is_valid_client_secret = $this->hasher->verifyHash($client_secret->getVerificationString(), $client_verification_expected_string);
        if (! $is_valid_client_secret) {
            throw new InvalidOAuth2AppSecretException($app);
        }

        return $app;
    }
}
