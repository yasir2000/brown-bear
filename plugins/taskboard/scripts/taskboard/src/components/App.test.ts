/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import App from "./App.vue";
import { createStoreMock } from "../../../../../../src/scripts/vue-components/store-wrapper-jest";
import type { ColumnDefinition } from "../type";
import type { RootState } from "../store/type";

describe("App", () => {
    function getStore(
        has_content: boolean,
        columns: Array<ColumnDefinition>,
        has_global_error: boolean,
        has_modal_error: boolean
    ): unknown {
        return createStoreMock({
            state: {
                has_content: has_content,
                error: {
                    has_global_error,
                    has_modal_error,
                },
                column: {
                    columns,
                },
            } as RootState,
        });
    }

    it("displays misconfiguration error when there are no column", () => {
        const wrapper = shallowMount(App, {
            mocks: { $store: getStore(true, [], false, false) },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    it("displays misconfiguration error even if there are no content", () => {
        const wrapper = shallowMount(App, {
            mocks: { $store: getStore(false, [], false, false) },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    it("displays the board when there are columns", () => {
        const wrapper = shallowMount(App, {
            mocks: {
                $store: getStore(
                    true,
                    [
                        { id: 2, label: "To do" },
                        { id: 3, label: "Done" },
                    ] as Array<ColumnDefinition>,
                    false,
                    false
                ),
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    it("displays empty state when there is no content", () => {
        const wrapper = shallowMount(App, {
            mocks: {
                $store: getStore(
                    false,
                    [
                        { id: 2, label: "To do" },
                        { id: 3, label: "Done" },
                    ] as Array<ColumnDefinition>,
                    false,
                    false
                ),
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it("displays global error state when there is an error", () => {
        const wrapper = shallowMount(App, {
            mocks: {
                $store: getStore(true, [], true, false),
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    describe(`mounted()`, () => {
        it(`will listen to the "keyup" event`, () => {
            jest.spyOn(document, "addEventListener").mockImplementation();

            shallowMount(App, {
                mocks: {
                    $store: getStore(true, [], false, false),
                },
            });

            expect(document.addEventListener).toHaveBeenCalledWith("keyup", expect.any(Function));
        });
    });

    describe(`destroy()`, () => {
        it(`will remove the "keyup" listener`, () => {
            jest.spyOn(document, "removeEventListener").mockImplementation();

            const wrapper = shallowMount(App, {
                mocks: {
                    $store: getStore(true, [], false, false),
                },
            });
            wrapper.destroy();

            expect(document.removeEventListener).toHaveBeenCalledWith(
                "keyup",
                expect.any(Function)
            );
        });
    });
});
