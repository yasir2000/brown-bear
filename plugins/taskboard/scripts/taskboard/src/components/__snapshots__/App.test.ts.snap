// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App displays empty state when there is no content 1`] = `
<div
  class="taskboard-app"
>
  <no-content-empty-state-stub />
</div>
`;

exports[`App displays global error state when there is an error 1`] = `
<div
  class="taskboard-app"
>
  <global-app-error-stub />
</div>
`;

exports[`App displays misconfiguration error even if there are no content 1`] = `
<div
  class="taskboard-app"
>
  <board-without-any-columns-error-stub />
</div>
`;

exports[`App displays misconfiguration error when there are no column 1`] = `
<div
  class="taskboard-app"
>
  <board-without-any-columns-error-stub />
</div>
`;

exports[`App displays the board when there are columns 1`] = `
<div
  class="taskboard-app"
>
  <task-board-stub />
</div>
`;
